Fix part of #14419: CATEGORY, ENTRY_POINT, STATS_PARAMETER_NAMES, PARAM_NAMES refactored to Category, EntryPoint, StatsParameterNames, ParamNames and values changed to UPPERCASE

Rostkowski opened the pull request:
Refactored following classes as per requirements of #14419.
CATEGORY,
ENTRY_POINT,
STATS_PARAMETER_NAMES,
PARAM_NAMES,

## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #14419 
2. This PR does the following: CATEGORY, ENTRY_POINT, STATS_PARAMETER_NAMES, PARAM_NAMES refactored to Category, EntryPoint, StatsParameterNames, ParamNames and values changed to UPPERCASE as per  #14419 requirements

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

No proof of changes needed because no new logic is introduced in the code base; only naming style of already tested classes is changed.


## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

Rostkowski committed:
Fix part of #14419

Refactored following classes as per requirements of #14419.
CATEGORY,
ENTRY_POINT,
STATS_PARAMETER_NAMES,
PARAM_NAMES,

Rostkowski committed:
Merge branch 'oppia:develop' into develop

seanlip commented on core/controllers/platform_feature.py:
```diff
@@ -101,7 +101,7 @@ def get(self):
         # visible when the dummy_feature is enabled.
         if not platform_feature_services.is_feature_enabled(
-                platform_feature_list.PARAM_NAMES.dummy_feature.value):
+                platform_feature_list.PARAM_NAMES.DUMMY_FEATURE.value):
```
Shouldn't PARAM_NAMES be UpperCamelCased too?

Rostkowski replied:
Got you! I checked again It should be good now.

Rostkowski committed:
Fix part of #14419

Fixed leftover PARAM_NAMES to ParamNames in core\controllers\platform_feature.py and  CATEGORY, ENTRY_POINT, STATS_PARAMETER_NAMES, PARAM_NAMES refactored to Category, EntryPoint, StatsParameterNames, ParamNames in docstrings

Rostkowski committed:
Merge branch 'oppia:develop' into fix-refactor

seanlip commented on core/domain/app_feedback_report_constants.py:
```diff
@@ -65,54 +65,54 @@ class ReportType(enum.Enum):
 # throughout the codebase according to the coding style guide.
 # https://github.com/oppia/oppia/wiki/Coding-style-guide
-class CATEGORY(enum.Enum): # pylint: disable=invalid-name
+class Category(enum.Enum): # pylint: disable=invalid-name
```
Why do you still need all the pylint disables here and below? Wouldn't these changes fix the issues?

Rostkowski replied:
Removed pylint disables.

Rostkowski committed:
Merge branch 'oppia:develop' into fix-refactor

Rostkowski committed:
Fix part of oppia#14419

Removed pylint disable and comments about todo

Rostkowski committed:
Merge branch 'fix-refactor' of https://github.com/Rostkowski/oppia into fix-refactor

seanlip commented on core/platform_feature_list.py:
```diff
@@ -21,7 +21,7 @@
 
 
-PARAM_NAMES = params.PARAM_NAMES # pylint: disable=invalid-name
+ParamNames = params.ParamNames # pylint: disable=invalid-name
```
Is this name still invalid? (cf the pylint pragma)

seanlip replied:
@Rostkowski Please don't resolve comments. It's important to follow the instructions [here](https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve), otherwise your PR review will encounter delays.

Also, just a note, I don't actually know whether the pylint pragma is needed here or not. Make sure to run the lint checks on your local machine to find out, if you haven't done so already! (Instructions for this should be in the wiki and also at the top of scripts/pre_commit_linter.py.)

Rostkowski committed:
Fix part of oppia#14419

Deleted leftover pylint disable.
